src/components/progress-report/workflow/progress-report-workflow.repository.ts

```diff
@@ -146,4 +152,79 @@ export class ProgressReportWorkflowRepository extends DtoRepository(
       changes: { status },
     });
   }
+
+  async getProjectMemberInfoByReportId(reportId: ID) {
+    const query = this.db
+      .query()
+      .match([
+        node('report', 'ProgressReport', { id: reportId }),
+        relation('in', '', ACTIVE),
+        node('engagement', 'Engagement'),
+        relation('in', '', 'engagement', ACTIVE),
+        node('project', 'Project'),
+        relation('out', '', 'member', ACTIVE),
+        node('member', 'ProjectMember'),
+        relation('out', '', 'user', ACTIVE),
+        node('user', 'User'),
+      ])
+      .optionalMatch([
+        node('user'),
+        relation('out', '', 'email', ACTIVE),
+        node('email', 'EmailAddress'),
+      ])
+      .match([
+        node('member'),
+        relation('out', '', 'roles', ACTIVE),
+        node('role', 'Property'),
+      ])
+      .return<{
+        id: ID;
+        email?: string;
+        role?: Role[];
```

Comment:
This is a nit, but var names for lists should be plural.

I don't the property value will ever be null, but you could also coalesce it to be sure.
```cql
coalesce(roles.value, []) as roles
```